src/operations.jl

```diff
@@ -5,25 +5,9 @@ hash(c::AbstractRGB, hx::UInt) = hash(blue(c), hash(green(c), hash(red(c), hx)))
 hash(c::TransparentRGB, hx::UInt) = hash(alpha(c), hash(blue(c), hash(green(c), hash(red(c), hx))))
 
 # rand
-typealias RandTypes{T,C} Union{AbstractRGB{T},
-                               TransparentRGB{C,T},
-                               AbstractGray{T},
-                               TransparentGray{C,T}}
-typealias RandTypesFloat{C,T<:AbstractFloat} RandTypes{T,C}
-
-rand{G<:AbstractGray}(::Type{G}) = G(rand())
-rand{G<:TransparentGray}(::Type{G}) = G(rand(), rand())
-rand{C<:AbstractRGB}(::Type{C}) = C(rand(), rand(), rand())
-rand{C<:TransparentRGB}(::Type{C}) = C(rand(), rand(), rand(), rand())
-function rand{C<:RandTypesFloat}(::Type{C}, sz::Dims)
-    reinterpret(C, rand(eltype(C), (sizeof(C)÷sizeof(eltype(C)), sz...)), sz)
-end
-function rand{C<:RandTypes{N0f8}}(::Type{C}, sz::Dims)
-    reinterpret(C, rand(UInt8, (sizeof(C), sz...)), sz)
-end
-function rand{C<:RandTypes{N0f16}}(::Type{C}, sz::Dims)
-    reinterpret(C, rand(UInt16, (sizeof(C)>>1, sz...)), sz)
-end
+rand{T<:Colorant}(::Type{T}) = reinterpret(T, rand(eltype(T), div(sizeof(T), sizeof(eltype(T)))))[1]
+rand{T<:Colorant}(::Type{T}, sz::Dims) =
+      reinterpret(T, rand(eltype(T), (div(sizeof(T), sizeof(eltype(T))), sz...)), sz)
```

Comment:
```julia
julia> using ColorTypes

julia> bja_rand{T<:Colorant}(::Type{T}) = reinterpret(T, rand(eltype(T), div(sizeof(T), sizeof(eltype(T)))))[1]
bja_rand (generic function with 1 method)

julia> bja_rand(RGB)
ERROR: argument is an abstract type; size is indeterminate
 in bja_rand(::Type{ColorTypes.RGB}) at ./REPL[2]:1

julia> rand(RGB)
RGB{Float64}(0.30389835762878525,0.1187732371310497,0.9873748322983826)
```
In the version I suggested, that `ccolor` call basically provides a "default fallback" of `Float64` (since that what `rand` actually produces).